perf/src/null.rs

```diff
@@ -0,0 +1,191 @@
+use std::sync::Arc;
+
+use bytes::BytesMut;
+
+use quinn_proto::{
+    crypto::{self, CryptoError},
+    transport_parameters, ConnectionId, Side, TransportError,
+};
+
+/// A rustls TLS session which does not perform packet encryption/decryption (for debugging purpose)
+struct NullCipherSession {
+    inner: Box<dyn crypto::Session>,
+}
+
+impl NullCipherSession {
+    fn new(tls: Box<dyn crypto::Session>) -> Self {
+        Self { inner: tls }
+    }
+}
+
+struct NullCipherPacketKey {
+    inner: Box<dyn crypto::PacketKey>,
+}
+
+impl NullCipherPacketKey {
+    fn new(key: Box<dyn crypto::PacketKey>) -> Self {
+        Self { inner: key }
+    }
+}
+
+pub struct NullCiperClientConfig {
+    inner: Arc<rustls::ClientConfig>,
+}
+
+impl NullCiperClientConfig {
+    pub fn new(config: Arc<rustls::ClientConfig>) -> Self {
+        Self { inner: config }
+    }
+}
+
+pub struct NullCiperServerConfig {
+    inner: Arc<rustls::ServerConfig>,
+}
+
+impl NullCiperServerConfig {
+    pub fn new(config: Arc<rustls::ServerConfig>) -> Self {
+        Self { inner: config }
```

Comment:
Why are we storing a rustls config in here??

Reply:
See references to `self.inner` below. In short, this logic creates a TLS session as usual but short-circuits the packet protection.

Reply:
Why don't we go whole hog and avoid all the TLS work?

Reply:
There is two reasons for this:
- we want to do everything TLS related (handshake, key rolling, ...) to see it in profile/flamegraph but we don't want the profile to be overwhelmed with the crypto operation
- doing like this require far less modification

What could also be bypassed is header protection and also packet tags as both of those operation also do some kind of crypto operations.

Reply:
Ah, okay, thanks.